src/elements/mixin/pipeaperture.H

```diff
@@ -60,42 +62,43 @@ namespace impactx::elements::mixin
             using namespace amrex::literals; // for _rt and _prt
 
             // skip aperture application if aperture_x <= 0 or aperture_y <= 0
-            if (m_aperture_x > 0 && m_aperture_y > 0) {
+            if (m_inv_aperture_x > 0_prt && m_inv_aperture_y > 0_prt) {
 
                // scale horizontal and vertical coordinates
-               amrex::ParticleReal const u = x / m_aperture_x;
-               amrex::ParticleReal const v = y / m_aperture_y;
+               amrex::ParticleReal const u = x * m_inv_aperture_x;
+               amrex::ParticleReal const v = y * m_inv_aperture_y;
 
                // compare against the aperture boundary
                if (amrex::Math::powi<2>(u) + amrex::Math::powi<2>(v) > 1_prt) {
                   amrex::ParticleIDWrapper{idcpu}.make_invalid();
                }
-
             }
         }
 
         /** Horizontal aperture size
          *
-         * @return horizontal aperture size in m
+         * @return horizontal aperture size [m]
          */
         AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
         amrex::ParticleReal aperture_x () const
         {
-            return m_aperture_x;
+            using namespace amrex::literals; // for _prt
+            return 1_prt / m_inv_aperture_x;
         }
 
         /** Vertical aperture size
          *
-         * @return vertical aperture size in m
+         * @return vertical aperture size [m]
          */
         AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
         amrex::ParticleReal aperture_y () const
         {
-            return m_aperture_y;
+            using namespace amrex::literals; // for _prt
+            return 1_prt / m_inv_aperture_y;
         }
 
-        amrex::ParticleReal m_aperture_x = 0; //! horizontal aperture size [m]
-        amrex::ParticleReal m_aperture_y = 0; //! vertical aperture size [m]
+        amrex::ParticleReal m_inv_aperture_x = 0; //! inverse of the horizontal aperture size [1/m]
```

Comment:
```suggestion
        // performance optimization: we intentionally store the inverse of
        //   the aperture, so we can do a quick multiplication instead of a costly
        //   division in the comparison operation per particle
        amrex::ParticleReal m_inv_aperture_x = 0; //! inverse of the horizontal aperture size [1/m]
```